prov/efa/src/rdm/rxr_pkt_pool.c

```diff
@@ -151,8 +180,13 @@ int rxr_pkt_pool_grow(struct rxr_pkt_pool *rxr_pkt_pool)
  */
 void rxr_pkt_pool_destroy(struct rxr_pkt_pool *pkt_pool)
 {
-	ofi_bufpool_destroy(pkt_pool->entry_pool);
+	if (pkt_pool->entry_pool)
+		ofi_bufpool_destroy(pkt_pool->entry_pool);
+
 	if (pkt_pool->sendv_pool)
 		ofi_bufpool_destroy(pkt_pool->sendv_pool);
+
+	free(pkt_pool->efa_send_wr_pool);
```

Comment:
free can be applied to NULL. 

The following is from manpage of `free`:

```
 The free() function frees the memory space pointed to by ptr, which must have been returned by a previous call to malloc(), calloc() or realloc().  Otherwise, or if free(ptr) has already been called before,
       undefined behavior occurs.  If ptr is NULL, no operation is performed.
```

Reply:
My concern was calling `free` when `send_wr_pool` wasn't allocated. You have a check for the `recv_wr_pool` right below. `rxr_pkt_pool` is allocated with `calloc`, so checking for NULL is same as checking for allocation.

Reply:
> My concern was calling free when send_wr_pool wasn't allocated.

When `send_wr_pool` is not allocated. It is NULL, and free(NULL) is ok.

> You have a check for the recv_wr_pool right below. 

I see. That is unnecessary, and should be removed

Reply:
Remove the check on `if (pool->recv_wr_pool)`